client/signup/steps/import-url/index.jsx

```diff
@@ -2,165 +2,279 @@
+		const {
+			isSiteImportableError,
+			goToNextStep,
+			urlInputValue,
+			stepName,
+			siteDetails,
+		} = this.props;
 
-	return isWebUri( withProtocol ) && hasDot;
-};
+		// isSiteImportable error, focus input to revise url.
+		if (
+			! isEqual( prevProps.isSiteImportableError, isSiteImportableError ) &&
+			isSiteImportableError
+		) {
+			this.focusInput();
+		}
 
-class ImportURLStepComponent extends Component {
-	componentDidMount() {
-		const { queryObject } = this.props;
-		const urlFromQueryArg = normalizeUrlForImportSource( get( queryObject, 'url' ) );
+		// We have a verified, importable site url.
+		if ( ! isEqual( prevProps.siteDetails, siteDetails ) && siteDetails ) {
+			SignupActions.submitSignupStep( { stepName }, [], {
+				importUrl: urlInputValue,
+				themeSlugWithRepo: 'pub/radcliffe-2',
+			} );
 
-		if ( urlFromQueryArg ) {
-			this.handleInputChange( urlFromQueryArg );
+			goToNextStep();
 		}
 	}
 
-	state = {
-		showValidation: false,
+	handleInputChange = event => {
+		this.props.setNuxUrlInputValue( event.target.value );
 	};
 
-	handleAction = () => {
-		if ( this.props.urlInputValidationMessage ) {
-			return this.setState( {
-				showValidation: true,
-			} );
+	handleInputBlur = event => {
+		if ( event.target.value ) this.validateUrl();
+	};
+
+	handleInputRef = el => ( this.inputRef = el );
+
+	focusInput = () => this.inputRef && this.inputRef.focus();
+
+	handleSubmit = event => {
+		event.preventDefault();
+
+		const isValid = this.validateUrl();
+
+		if ( ! isValid ) {
+			this.focusInput();
+			return;
 		}
 
-		this.goToNextStep( {} );
+		this.props.fetchIsSiteImportable( this.props.urlInputValue );
 	};
 
-	goToNextStep = () => {
-		const { urlInputValue } = this.props;
-		SignupActions.submitSignupStep( { stepName: this.props.stepName }, [], {
-			importUrl: urlInputValue,
-			themeSlugWithRepo: 'pub/radcliffe-2',
+	validateUrl = () => {
+		const validationMessage = validateImportUrl( this.props.urlInputValue );
+		const isValid = ! validationMessage;
+
+		this.setState( {
+			urlValidationMessage: isValid ? '' : validationMessage,
+			showUrlMessage: true,
 		} );
 
-		this.props.goToNextStep();
+		return isValid;
 	};
 
-	handleInputChange = value => {
-		const validationMessage = isValidUrl( value )
-			? ''
-			: this.props.translate( 'Please enter a valid URL.' );
+	getUrlMessage = () => {
+		if ( this.state.urlValidationMessage ) {
+			return this.state.urlValidationMessage;
+		} else if ( this.props.isSiteImportableError ) {
+			return this.getIsSiteImportableError();
+		}
 
-		this.setState( {
-			showValidation: false,
-		} );
+		return '';
+	};
 
-		this.props.setNuxUrlInputValue( value );
-		this.props.setValidationMessage( validationMessage );
+	getIsSiteImportableError = () => {
+		if ( ! this.props.isSiteImportableError ) {
+			return null;
+		}
+
+		const { isSiteImportableError, translate } = this.props;
+
+		switch ( isSiteImportableError.code ) {
+			case 1000001:
```

Comment:
These should be enumerated constants that this component imports.

Reply:
Excellent idea. Updated.